### ISSUE TITLE
Requests with params as `none` or `void` should not require a `nil` return

### ISSUE THREAD

**Issue by wkirschbaum** [elixir-tools/gen_lsp#30]
Requests like `GenLSP.Requests.Shutdown' and `GenLSP.Requests.WorkspaceCodeLensRefresh' might receive empty params or no params as nil, which raises the follow error when the client returns empty params: 

```bash
LSP Exited.

Last message received: handle_notification %{"jsonrpc" => "2.0", "method" => "initialized", "params" => %{}}

** (MatchError) no match of right hand side value: {:error, %{"params" => "expected nil"}}
    (gen_lsp 0.3.0) lib/gen_lsp.ex:228: anonymous fn/5 in GenLSP.loop/3
    (gen_lsp 0.3.0) lib/gen_lsp.ex:292: GenLSP.attempt/3
    (stdlib 5.0.2) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
``` 

Between the spec version 3.16 and 3.17 they changed these from void to none. Removing he requirement for params being nil seems to be more appropriate.

Currently a client for Emacs, lsp-mode spits out errors because of this.

**Comment by wkirschbaum** [comment]
Setting the nil to any() seems to fix the issue:

```diff
@@ -15,7 +15,7 @@ defmodule GenLSP.Notifications.Exit do
   typedstruct do
     field :method, String.t(), default: "exit"
     field :jsonrpc, String.t(), default: "2.0"
-    field :params, nil
+    field :params, any()
   end

  ```

**Comment by wkirschbaum** [comment]
The following seems to work: 

```diff
@@ -18,7 +18,7 @@ defmodule GenLSP.Requests.Shutdown do
     field :method, String.t(), default: "shutdown"
     field :jsonrpc, String.t(), default: "2.0"
     field :id, integer(), enforce: true
-    field :params, nil
+    field :params, any()
   end
 
   @type result :: nil
@@ -30,7 +30,7 @@ defmodule GenLSP.Requests.Shutdown do
       method: "shutdown",
       jsonrpc: "2.0",
       id: int(),
-      params: nil
+      params: any()
     })
   end
```

**Comment by mhanberg** [comment]
You will need to make the change in the codegen repo and then re generate the files in this repo. There is a bin script to do it

**Comment by mhanberg** [comment]
Closed by #32
